src/components/shared/SgtIButton.vue

```diff
@@ -0,0 +1,91 @@
+<!--
```

Comment:
can you name this component as something else so that you could change the icon so can take as input prop

Reply:
I have refactored the component to accept different icons now.